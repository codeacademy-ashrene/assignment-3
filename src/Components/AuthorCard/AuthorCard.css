.authorName{
  margin: 2%;
}
fieldset{
  border: solid;
  border-color: orange;
  border-radius: 30px;
  border-width: 5px;
}
.authorCard{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 5%;
  
  
}

.author{
  background: white;
  width: 20%;
  margin:auto;
  padding: 1%;
  color: orange;
  font-style: italic;
  font-size: 20px;
  text-align: center;
  font-weight: bold;
  box-sizing: border-box;
  overflow: hidden;
  min-width: 100px;
  /* z-index: 10;
  position: absolute;
  left: 50% */
  
}