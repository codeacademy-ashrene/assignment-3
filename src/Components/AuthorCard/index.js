import React, { Component } from 'react';
import './AuthorCard.css';
import BookCard from '../BookCard';

class AuthorCard extends Component {
  render() {
    const cardRow = [];
    let author;
    let counter = 0;
    this.props.bookData.forEach((element) => {
      counter += 1;
      author = element.Author;
      cardRow.push(
        <BookCard bookData={element} styleCounter={counter} key={element.id} />,
      );
    });
    return (
      <div className="authorName">
        <fieldset>
          <legend className="author">{author}</legend>
          <div className="authorCard">
            {cardRow}
          </div>
        </fieldset>

      </div>
    );
  }
}

export default AuthorCard;
