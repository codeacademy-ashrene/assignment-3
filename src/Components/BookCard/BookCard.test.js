import React from 'react';
import renderer from 'react-test-renderer';
import AuthorCard from '.';


describe('AuthorCard',()=>{
    const bookData={
        Author: 'J K Rowling',
        id: 10,
        Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)',
        rating: 4.45
      }
    
    it('should render header',()=>{
        const tree=renderer.create(<AuthorCard bookData={bookData}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});