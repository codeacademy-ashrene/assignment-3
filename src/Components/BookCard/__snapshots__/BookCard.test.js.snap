// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorCard should render header 1`] = `
<div
  className="card"
>
  <div
    className="cardContentOrange"
  >
    <img
      alt="Harry Potter Books"
      className="image"
      src="./cover_image.jpeg"
    />
    <div
      className="cardTextAndReaction"
    >
      <div
        className="cardText"
      >
        Harry Potter and the Sorcerers Stone (Harry Potter, #1)
        10
      </div>
      <div
        className="rating"
      >
        4.5
        <img
          alt="heart"
          className="heart"
          onClick={[Function]}
          src="./Icons/heart-plain.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
